### ISSUE TITLE
using apt filter settings results in warnings on the console for unsupported config settings

### ISSUE THREAD

**Issue by reubenmiller** [thin-edge/thin-edge.io#2109]
**Describe the bug**

Configuration apt filter parameters in the `/etc/tedge/tedge.toml` file causes warning messages to be displayed when calling `tedge config list`. Also, if the `tedge config set` command is used to change an unrelated field, it will result in the `[apt]` setting from being removed.

The apt filter settings were set according to the new feature available in 0.12.0, and documented [here](https://thin-edge.github.io/thin-edge.io/operate/installation/install_and_enable_software_management/#filter-packages-by-name-and-maintainer)

**To Reproduce**

1. Add the following section to the `/etc/tedge/tedge.toml` file.

    *file: /etc/tedge/tedge.toml*

    ```toml
    [apt]
    name = "(tedge|c8y|python|wget|vim|curl|apt|mosquitto|ssh|sudo).*"
    ```

2. List the config via the tedge cli

    ```sh
    tedge config list
    ```

    The following erroneous warning is printed to the console.

    ```text
    2023-07-28T10:31:09.725262676Z  WARN tedge_config::tedge_config_cli::figment: Unknown configuration field "apt" from TOML file /etc/tedge/tedge.toml
    ```

    **Note**

    The filter function does work as expected, it is only the `tedge config` commands which do not currently recognize the settings.

3. Update another setting via the `tedge config set`

    ```sh
    tedge config set c8y.url "thin-edge-io.eu-latest.cumulocity.com"
    ```

4. View the `/etc/tedge/tedge.toml` setting again. The `[apt]` section would have been removed erroneously.

**Expected behavior**

* No warning message should be printed by thin-edge components for officially documented features.
* The `[apt]` section should not be removed if other settings are updated

**Screenshots**

**Environment (please complete the following information):**
 - OS [incl. version]: Debian 11
 - Hardware [incl. revision]: Docker running on aarch64
 - System-Architecture [e.g. result of "uname -a"]: `Linux 00765dbcfe45 5.15.68-0-virt #1-Alpine SMP Fri, 16 Sep 2022 06:29:31 +0000 aarch64 GNU/Linux`
 - thin-edge.io version [e.g. 0.1.0]: 0.12.0

**Additional context**

I suspect the root cause is that the tedge configuration object just does not respect/recognize the `[apt]` section which causes the warning and the remove of it when any other managed setting is updated/removed.

**Comment by reubenmiller** [comment]
Available to be tested >= 0.12.0-22-g052f9dc3
Merged in build: https://github.com/thin-edge/thin-edge.io/actions/runs/5739700529

**Comment by gligorisaev** [comment]
QA has thoroughly checked the feature and here are the results:

- [x] Test for ticket exists in the test suite.  
         /tests/RobotFramework/tests/tedge/call_tedge_config_list.robot
- [x] QA has tested the feature and it meets the required specifications.
